include/secp256k1_silentpayments.h

```diff
@@ -0,0 +1,451 @@
+ *                            key, a spend public key, and an index indicating
+ *                            its position in the original ordering. The
+ *                            recipient array will be grouped by scan public key
+ *                            in place (as specified in BIP0352), but generated
+ *                            outputs are saved in the `generated_outputs` array
+ *                            to match the original ordering (using the index
+ *                            field). This ensures the caller is able to match
+ *                            the generated outputs to the correct silent
+ *                            payment addresses. The same recipient can be
+ *                            passed multiple times to create multiple outputs
+ *                            for the same recipient.
+ *              n_recipients: the number of recipients. This is equal to the
+ *                            total number of outputs to be generated as each
+ *                            recipient may passed multiple times to generate
+ *                            multiple outputs for the same recipient
+ *       outpoint_smallest36: serialized (36-byte) smallest outpoint
+ *                            (lexicographically) from the transaction inputs
+ *           taproot_seckeys: pointer to an array of pointers to taproot
+ *                            keypair inputs (can be NULL if no secret keys
+ *                            of taproot inputs are used)
+ *         n_taproot_seckeys: the number of sender's taproot input secret keys
+ *             plain_seckeys: pointer to an array of pointers to 32-byte
+ *                            secret keys of non-taproot inputs (can be NULL
+ *                            if no secret keys of non-taproot inputs are
+ *                            used)
+ *           n_plain_seckeys: the number of sender's non-taproot input secret
+ *                            keys
+ */
+SECP256K1_API SECP256K1_WARN_UNUSED_RESULT int secp256k1_silentpayments_sender_create_outputs(
+    const secp256k1_context *ctx,
+    secp256k1_xonly_pubkey **generated_outputs,
+    const secp256k1_silentpayments_recipient **recipients,
+    size_t n_recipients,
+    const unsigned char *outpoint_smallest36,
+    const secp256k1_keypair * const *taproot_seckeys,
+    size_t n_taproot_seckeys,
+    const unsigned char * const *plain_seckeys,
+    size_t n_plain_seckeys
+) SECP256K1_ARG_NONNULL(1) SECP256K1_ARG_NONNULL(2) SECP256K1_ARG_NONNULL(3) SECP256K1_ARG_NONNULL(5);
+
+/** Create Silent Payment label tweak and label.
+ *
+ *  Given a recipient's 32 byte scan key and a label integer m, calculate the
+ *  corresponding label tweak and label:
+ *
+ *      label_tweak = hash(scan_key || m)
+ *            label = label_tweak * G
+ *
+ *  Returns: 1 if label tweak and label creation was successful.
+ *           0 if label_tweak32 is an invalid scalar,
```

Comment:
```suggestion
 *           0 if hash output label_tweak32 is not valid scalar (negligible
 *             probability per hash evaluation)
```
would be more consistent with my other suggested changes regarding docs and implies simpler error handling (because this should never happen).